packages/snap-preact/components/src/hooks/createRecommendationTemplate.tsx

```diff
@@ -0,0 +1,77 @@
+import { RecommendationController, RecommendationControllerConfig } from '@searchspring/snap-controller';
+import { SnapTemplates } from '../../../src';
+import { Theme, useSnap } from '../providers';
+import { useCreateController } from './useCreateController';
+import { useComponent } from './useComponent';
+import { ResultComponent, RecommendationComponentNames } from '../types';
+import { FunctionalComponent } from 'preact';
+import { RecommendationProps } from '../components/Templates/Recommendation';
+import { RecommendationGridProps } from '../components/Templates/RecommendationGrid';
+
+export type templatesType = {
+	recommendation?: {
+		enabled: boolean;
+		component?: RecommendationComponentNames; // Need a type for allowed recommendation component names (that would exist in the library)
+		resultComponent?: string;
+		config?: Partial<RecommendationControllerConfig>;
+	};
+};
+
+type returnType = {
+	RecommendationTemplateComponent: FunctionalComponent<RecommendationProps | RecommendationGridProps> | undefined;
+	RecommendationTemplateResultComponent: ResultComponent | undefined;
+	recsController: RecommendationController | undefined;
+};
+
+export function createRecommendationTemplate(templates: templatesType, theme?: Theme): returnType {
+	let recommendationTemplateComponent: FunctionalComponent<{ controller: RecommendationController; name: string }> | undefined;
+	let recommendationTemplateResultComponent: ResultComponent | undefined;
+	let recsController: RecommendationController | undefined;
+
+	if (templates?.recommendation?.enabled) {
+		const componentName = templates?.recommendation?.component || 'Recommendation';
```

Comment:
I wonder if the default should be `RecommendationGrid` - if so we might want to fix this:
<img width="796" alt="image" src="https://github.com/user-attachments/assets/7f2b7904-b588-48b3-b657-3aba49ee7245" />

The title should be outside of the scrollable div (if possible).